crates/egui/src/data/input.rs

```diff
@@ -176,8 +176,11 @@ pub enum ViewportEvent {
 #[derive(Clone, Debug, Default, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 pub struct ViewportInfo {
+    /// this viewport, if known.
+    pub this: Option<ViewportId>,
```

Comment:
This is unused?

I think everywhere we have a `ViewportInfo` we have a `ViewportId` for it as the key

Reply:
This is used.
Since parent_viewport_id() returns the last viewport_id, it is difficult to use unless it is inside a crate.

Reply:
> This is unused?
> 
> I think everywhere we have a `ViewportInfo` we have a `ViewportId` for it as the key

no. There are places where viewport_id is missing. It's hard to explain.

Reply:
> Try to explain and/or make this a `ids: ViewportIdPair` which is set in a constructor, instead of sometimes being `None`.

There are parts where this is absolutely necessary, and users will be able to easily access it. (In fact, this may not be easy for beginners to access.)
This is absolutely necessary, as there are cases where there is currently no way to determine the viewport_id.
Without this, every time you create a viewport, you would have to save the viewport_id somewhere and take care of it.

Reply:
> Try to explain and/or make this a `ids: ViewportIdPair` which is set in a constructor, instead of sometimes being `None`.

If you don't save the viewport_id when creating a viewport, the current window sometimes doesn't know what the current viewport_id is. This is difficult to explain. Currently the viewport_id is unknown, so how can I find the parent_id? This is a different case from examples/demo_viewports.
last viewport_id may not be the current viewport.

Reply:
> Then add `ids: ViewportIdPair` to `ViewportInfo` (and remove the `Default` bound on it), so users can _depend_ on the ids

`info.this` is absolutely necessary.
If `info.this` is present, `info.parent` does not need to be present.
`info.parent` pre-existed and could not work.